packages/fuels-contract/src/contract.rs

```diff
@@ -45,6 +43,56 @@ pub struct CallResponse<D> {
     pub receipts: Vec<Receipt>,
 }
 
+pub struct ContractCallArguments {
+    contract_id: ContractId,
+    encoded_selector: Option<Selector>,
+    encoded_args: Option<Vec<u8>>,
+    tx_parameters: TxParameters,
+    call_parameters: CallParameters,
+    variable_outputs: Option<Vec<Output>>,
+    maturity: Word,
+    compute_calldata_offset: bool,
+    external_contracts: Option<Vec<ContractId>>,
+    simulate: bool,
+}
+impl ContractCallArguments {
+    pub fn new(
+        contract_id: ContractId,
+        encoded_selector: Option<Selector>,
+        encoded_args: Option<Vec<u8>>,
+        variable_outputs: Option<Vec<Output>>,
+        compute_calldata_offset: bool,
+        external_contracts: Option<Vec<ContractId>>,
+    ) -> Self {
+        Self {
+            contract_id,
+            encoded_selector,
+            encoded_args,
+            tx_parameters: TxParameters::default(),
+            call_parameters: CallParameters::default(),
+            variable_outputs,
+            maturity: DEFAULT_MATURITY as Word,
+            compute_calldata_offset,
+            external_contracts,
+            // TODO: the fact that transactions are not simulated by default should be a flag
+            simulate: false,
+        }
+    }
+
+    pub fn set_tx_parameters(&mut self, tx_params: TxParameters) {
+        self.tx_parameters = tx_params
+    }
+    pub fn set_call_parameters(&mut self, call_params: CallParameters) {
+        self.call_parameters = call_params
+    }
+    pub fn set_maturity(&mut self, maturity: Word) {
+        self.maturity = maturity
+    }
+    pub fn set_simulate(&mut self, simulate: bool) {
+        self.simulate = simulate
+    }
```

Comment:
Hmmm. Do we really want this to be set like the other parameters? It's neither a call not a tx parameter. It's a separate thing, about what the SDK will actually do.

Reply:
My idea is that "by default", you usually don't simulate, so having to do `set_simulate` isn't a big deal. I can change it if you think that's not good.